doc/user-guide/io.rst

```diff
@@ -228,83 +234,37 @@ to the original netCDF file, regardless if they exist in the original dataset.
-:py:meth:`Dataset.to_netcdf` to ensure that each call does not delete the file.
-For example:
-
-.. ipython::
-    :verbatim:
-
-    In [1]: ds1 = xr.Dataset({"a": 0})
-
-    In [2]: ds2 = xr.Dataset({"b": 1})
-
-    In [3]: ds1.to_netcdf("file.nc", group="A")
-
-    In [4]: ds2.to_netcdf("file.nc", group="B", mode="a")
-
-We can verify that two groups have been saved using the ncdump command-line utility.
-
-.. code:: bash
-
-    $ ncdump file.nc
-    netcdf file {
-
-    group: A {
-      variables:
-        int64 a ;
-      data:
-
-       a = 0 ;
-      } // group A
-
-    group: B {
-      variables:
-        int64 b ;
-      data:
-
-       b = 1 ;
-      } // group B
-    }
-
-Either of these groups can be loaded from the file as an independent :py:class:`Dataset` object:
-
-.. ipython::
-    :verbatim:
-
-    In [1]: group1 = xr.open_dataset("file.nc", group="A")
+Whilst netCDF groups can only be loaded individually as ``Dataset`` objects, a
+whole file of many nested groups can be loaded as a single
+:py:class:`xarray.DataTree` object. To open a whole netCDF file as a tree of groups
+use the :py:func:`xarray.open_datatree` function. To save a DataTree object as a
+netCDF file containing many groups, use the :py:meth:`xarray.DataTree.to_netcdf` method.
 
-    In [2]: group1
-    Out[2]:
-    <xarray.Dataset>
-    Dimensions:  ()
-    Data variables:
-        a        int64 ...
 
-    In [3]: group2 = xr.open_dataset("file.nc", group="B")
-
-    In [4]: group2
-    Out[4]:
-    <xarray.Dataset>
-    Dimensions:  ()
-    Data variables:
-        b        int64 ...
+.. _netcdf.root_group.note:
 
 .. note::
+    Due to file format specifications the on-disk root group name is always ``"/"``,
+    overriding any given ``DataTree`` root node name.
 
-    For native handling of multiple groups with xarray, including I/O, you might be interested in the experimental
-    `xarray-datatree <https://github.com/xarray-contrib/datatree>`_ package.
+.. _netcdf.group.warning:
 
+.. warning::
+    ``DataTree`` objects do not follow the exact same data model as netCDF
+    files, which means that perfect round-tripping is not always possible.
+
+    In particular in the netCDF data model dimensions are entities that can
+    exist regardless of whether any variable possesses them. This is in contrast
+    to `xarray's data model <https://docs.xarray.dev/en/stable/user-guide/data-structures.html>`_
+    (and hence :ref:`DataTree's data model <data structures>`) in which the
+    dimensions of a (Dataset/Tree) object are simply the set of dimensions
+    present across all variables in that dataset.
+
+    This means that if a netCDF file contains dimensions but no variables which
+    possess those dimensions, these dimensions will not be present when that
+    file is opened as a DataTree object.
+    Saving this DataTree object to file will therefore not preserve these
+    "unused" dimensions.
```

Comment:
This may also be a good place to mention xarray's non-support for coordinates that conflict between parent and child nodes, and pointing users to `open_datasets_as_groups` as the recommended way to open such datasets.

Reply:
I'm not sure this is the right place for that.  This is under modifying existing zarr stores. They will always be aligned based on their data model?

Reply:
I think the point here is that somewhere in the IO docs for netCDF and Zarr there should be info or links to info about how open_datatree is not guaranteed to roundtrip but open_groups is.